### PR TITLE
Apply deprecated `evaluation_strategy`

### PR DESCRIPTION
**Pull request by muellerzr**
This PR applies the deprecated `evaluation_strategy` from https://github.com/huggingface/transformers/pull/30190.

TL;DR: `evaluation_strategy` -> `eval_strategy`

(Should go live in v4.41.0, but opening now so it's here when ready!)

### CODE REVIEW

**Review by lewtun — APPROVED** [review]
Thanks for the fix LGTM!

File: tests/test_dpo_trainer.py [hunk 1]
```diff
@@ -624,11 +624,11 @@ def test_dpo_lora_force_use_ref(self):
             training_args = DPOConfig(
                 output_dir=tmp_dir,
                 per_device_train_batch_size=2,
-                max_steps=3,
+                max_steps"=3,
```
Comment (kashif) [inline tests/test_dpo_trainer.py:627]:
```suggestion
                max_steps=3,
```

**Comment by younesbelkada** [comment]
@muellerzr can you rebase with main before merging?

**Comment by kashif** [comment]
@younesbelkada already rebased and fixed merge conflicts

**Pull request merged**